### PR TITLE
[ARG task]run agent service as user. Creating a new branch with the previous changes

### PR DESCRIPTION
**Pull request by tejasd1990**

### CODE REVIEW

File: Tasks/AzureResourceGroupDeploymentV2/task.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 2,
-        "Minor": 134,
+        "Minor": 135,
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeploymentV2/task.json]: This is sprint based. So you can name it 136

File: Tasks/AzureResourceGroupDeploymentV2/task.json [hunk 2]
```diff
@@ -240,6 +240,35 @@
...
+        },
+        {
+            "name": "password",
+            "type": "string",
+            "label": "Password",
+            "groupName": "AdvancedDeploymentOptions",
+            "visibleRule": "enableDeploymentPrerequisites = ConfigureVMWithDGAgent && runAgentServiceAsUser = true",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "The password for the user to run the agent service on the Windows VMs. It is assumed that the password is the same for the specified user on all the VMs. For linux VMs, password is not required and will be ignored."
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeploymentV2/task.json]: Should you have the text indicating the store the value as a secret variable

File: Tasks/AzureResourceGroupDeploymentV2/operations/DeploymentGroupExtensionHelper.ts [hunk 3]
```diff
@@ -206,13 +206,21 @@ export class DeploymentGroupExtensionHelper {
...
-                MachineGroup: this.taskParameters.deploymentGroupName,
                 DeploymentGroup: this.taskParameters.deploymentGroupName,
                 AgentName: "",
                 Tags: tags
             };
             console.log("Public settings are:\n VSTSAccountName: %s\nTeamProject: %s\nDeploymentGroup: %s\nTags: %s\n", collectionUri, teamProject, this.taskParameters.deploymentGroupName, JSON.stringify(tags));
             var protectedSettings = { PATToken: this.taskParameters.tokenCredentials.getPatToken() };
+            if (this.taskParameters.runAgentServiceAsUser) {
+                if (vmOsType === "Windows") {
+                    publicSettings["WindowsLogonAccountName"] = this.taskParameters.agentServiceUserCredentials.getUserName();
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeploymentV2/operations/DeploymentGroupExtensionHelper.ts]: why do we have a different parameter for windows and linux

File: Tasks/AzureResourceGroupDeploymentV2/task.json (hunk 1 above)
Comment (tejasd1990) [reply Tasks/AzureResourceGroupDeploymentV2/task.json]: Done

File: Tasks/AzureResourceGroupDeploymentV2/operations/DeploymentGroupExtensionHelper.ts (hunk 3 above)
Comment (tejasd1990) [reply Tasks/AzureResourceGroupDeploymentV2/operations/DeploymentGroupExtensionHelper.ts]: Issue added in vm extension repo

**Pull request merged**